gems/aws-sdk-core/spec/aws/instance_profile_credentials_spec.rb

```diff
@@ -379,39 +362,38 @@ module Aws
         expect(creds.account_id).to be_nil
       end
 
-      it 'Test IMDS credentials provider when account ID is unavailable when profile is unstable returns valid credentials' do
-        stub_request(:get, "#{ipv4_endpoint}#{extended_path}")
-          .to_return(status: 200, body: "my-profile-0007\n")
-          .to_return(status: 200, body: "my-profile-0007-b\n")
+      it 'Test IMDS credentials provider when account ID is unavailable'\
+         'when profile is unstable returns valid credentials' do
         resp1 = {
-          "Code": "Success",
-          "LastUpdated": "2025-03-18T20:53:17.832308Z",
-          "Type": "AWS-HMAC",
-          "AccessKeyId": "ASIAIOSFODNN7EXAMPLE",
-          "SecretAccessKey": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
-          "Token": "AQoEXAMPLEH4aoAH0gNCAPyJxz4BlCFFxWNE1OPTgk5TthT+FvwqnKw...(truncated)",
+          "Code": 'Success',
+          "LastUpdated": '2025-03-18T20:53:17.832308Z',
+          "Type": 'AWS-HMAC',
+          "AccessKeyId": 'ASIAIOSFODNN7EXAMPLE',
+          "SecretAccessKey": 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY',
+          "Token": 'AQoEXAMPLEH4aoAH0gNCAPyJxz4BlCFFxWNE1OPTgk5TthT+FvwqnKw...(truncated)',
           "Expiration": expiration,
           "UnexpectedElement7": {
-            "Name": "ignore-me-7"
+            "Name": 'ignore-me-7'
           }
         }
-        stub_request(:get, "#{ipv4_endpoint}#{extended_path}my-profile-0007")
-          .to_return(status: 200, body: resp1.to_json)
-          .to_return(status: 404)
         resp2 = {
-          "Code": "Success",
-          "LastUpdated": "2025-03-18T20:53:17.832308Z",
-          "Type": "AWS-HMAC",
-          "AccessKeyId": "ASIAIOSFODNN7EXAMPLE",
-          "SecretAccessKey": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
-          "Token": "AQoEXAMPLEH4aoAH0gNCAPyJxz4BlCFFxWNE1OPTgk5TthT+FvwqnKw...(truncated)",
+          "Code": 'Success',
+          "LastUpdated": '2025-03-18T20:53:17.832308Z',
+          "Type": 'AWS-HMAC',
+          "AccessKeyId": 'ASIAIOSFODNN7EXAMPLE',
+          "SecretAccessKey": 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY',
+          "Token": 'AQoEXAMPLEH4aoAH0gNCAPyJxz4BlCFFxWNE1OPTgk5TthT+FvwqnKw...(truncated)',
           "Expiration": expiration,
           "UnexpectedElement7": {
-            "Name": "ignore-me-7"
+            "Name": 'ignore-me-7'
           }
         }
-        stub_request(:get, "#{ipv4_endpoint}#{extended_path}my-profile-0007-b")
-          .to_return(status: 200, body: resp2.to_json)
+        stub_request(:get, metadata_uri)
```

Comment:
This is fine if you want - I ordered these stubs in sequence. Maybe consider more meaningful names instead of resp1 and resp2.